### PR TITLE
Add documentation for apollo-ast

### PR DESCRIPTION
**Pull request by BoD**
Related to #3490

### CODE REVIEW

File: apollo-ast/build.gradle.kts [hunk 1]
```diff
@@ -20,12 +20,5 @@ configurations[JavaPlugin.API_CONFIGURATION_NAME].let { apiConfiguration ->
 }
 
 tasks.withType(KotlinCompile::class.java) {
-  kotlinOptions {
```
Comment (BoD) [inline apollo-ast/build.gradle.kts:23]: Not completely sure if this is really no longer useful!

Comment (martinbonnin) [reply apollo-ast/build.gradle.kts:23]: It's no longuer useful if we merge https://github.com/apollographql/apollo-android/pull/3542

File: docs/source/advanced/apollo-ast.mdx [hunk 2]
```diff
@@ -0,0 +1,187 @@
+---
+title: Apollo AST
+---
+
+In order to generate code, Apollo Android reads your schema and operations written in the GraphQL language. To do this,
+we built a GraphQL parser: a tool that can read, interpret, and expose GraphQL language constructs in a way that can be
+used by the library.
```
Comment (martinbonnin) [inline docs/source/advanced/apollo-ast.mdx:7]: Nit: the rest of the doc usually uses a more imperative tone:

```suggestion
In order to generate code, Apollo Android reads your schema and operations written in the GraphQL language. To do this,
it parses your GraphQL files in an Abstract Syntax Tree ([AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree)). The AST represents your GraphQL document in a type-safe, machine-readable way.
```

File: apollo-ast/build.gradle.kts (hunk 1 above)
Comment (BoD) [reply apollo-ast/build.gradle.kts]: Ok, reverting for now :)

File: docs/source/advanced/apollo-ast.mdx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+Since this parser is not tied to the rest of the library, and as it may be useful for other purposes, we made it
+available as its own artifact that you can use in your projects.
```
Comment (martinbonnin) [inline docs/source/advanced/apollo-ast.mdx:10]:
```suggestion
This parser is in its own module and you can use independently of apollo-runtime or apollo-api. 
```

File: docs/source/advanced/apollo-ast.mdx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+Features:
+
+- Parses Schema and Operations
+- Returns an Abstract Syntax Tree (AST)
+- AST can easily be manipulated (`transform` API)
+- AST can be translated back to valid GraphQL syntax
```
Comment (martinbonnin) [inline docs/source/advanced/apollo-ast.mdx:17]:
```suggestion
- AST can be printed back to valid, indented GraphQL document
```

File: docs/source/advanced/apollo-ast.mdx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+- Validation of schema and operations exposing warnings (e.g. use of deprecated fields) and errors
```
Comment (martinbonnin) [inline docs/source/advanced/apollo-ast.mdx:18]:
```suggestion
- Validation rules for operations (See [GraphQL spec](https://spec.graphql.org/draft/#sec-Validation))
```

I wouldn't mention schema validation as we're pretty barebones there

File: docs/source/advanced/apollo-ast.mdx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+dependencies {
+  // ...
+
+  implementation("com.apollographql.apollo3:apollo-ast:x.y.z")
+}
+```
+
+## Parse a document
+
+Use the `parseAsGQLDocument` method to parse a document from a File, a String, or an
```
Comment (martinbonnin) [inline docs/source/advanced/apollo-ast.mdx:34]: ~Technically speaking, `parseAsGQLDocument` also does validation. Maybe we should change the name there.~ 

Oopps no, my bad, it's only parsing 👍

**Review by martinbonnin — APPROVED** [review]

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**